test/cookie/global-headers.js

```diff
@@ -16,55 +17,47 @@ if (!globalThis.Headers) {
   process.exit(0)
 }
 
-test('Using global Headers', (t) => {
-  t.test('deleteCookies', (t) => {
+test('Using global Headers', async (t) => {
+  await t.test('deleteCookies', (t) => {
     const headers = new Headers()
 
-    t.equal(headers.get('set-cookie'), null)
+    assert.equal(headers.get('set-cookie'), null)
     deleteCookie(headers, 'undici')
-    t.equal(headers.get('set-cookie'), 'undici=; Expires=Thu, 01 Jan 1970 00:00:00 GMT')
-
-    t.end()
+    assert.equal(headers.get('set-cookie'), 'undici=; Expires=Thu, 01 Jan 1970 00:00:00 GMT')
   })
 
   t.test('getCookies', (t) => {
     const headers = new Headers({
       cookie: 'get=cookies; and=attributes'
     })
 
-    t.same(getCookies(headers), { get: 'cookies', and: 'attributes' })
-    t.end()
+    assert.deepEqual(getCookies(headers), { get: 'cookies', and: 'attributes' })
   })
 
-  t.test('getSetCookies', (t) => {
+  await t.test('getSetCookies', (t) => {
     const headers = new Headers({
       'set-cookie': 'undici=getSetCookies; Secure'
     })
 
     const supportsCookies = getHeadersList(headers).cookies
 
     if (!supportsCookies) {
-      t.same(getSetCookies(headers), [])
+      assert.deepEqual(getSetCookies(headers), [])
     } else {
-      t.same(getSetCookies(headers), [
+      assert.deepEqual(getSetCookies(headers), [
         {
           name: 'undici',
           value: 'getSetCookies',
           secure: true
         }
       ])
     }
-
-    t.end()
   })
 
-  t.test('setCookie', (t) => {
+  await t.test('setCookie', (t) => {
```

Comment:
why await? this test is not async 🤔

Reply:
It's an alternative way to `describe` / `test` :

```
/**
     * The `test()` function is the value imported from the `test` module. Each
     * invocation of this function results in reporting the test to the `TestsStream`.
     *
     * The `TestContext` object passed to the `fn` argument can be used to perform
     * actions related to the current test. Examples include skipping the test, adding
     * additional diagnostic information, or creating subtests.
     *
     * `test()` returns a `Promise` that resolves once the test completes.
     * if `test()` is called within a `describe()` block, it resolve immediately.
     * The return value can usually be discarded for top level tests.
     * However, the return value from subtests should be used to prevent the parent
     * test from finishing first and cancelling the subtest
     * as shown in the following example.
     *
     * ```js
     * test('top level test', async (t) => {
     *   // The setTimeout() in the following subtest would cause it to outlive its
     *   // parent test if 'await' is removed on the next line. Once the parent test
     *   // completes, it will cancel any outstanding subtests.
     *   await t.test('longer running subtest', async (t) => {
     *     return new Promise((resolve, reject) => {
     *       setTimeout(resolve, 1000);
     *     });
     *   });
     * });
     * ```
     *
     * The `timeout` option can be used to fail the test if it takes longer than`timeout` milliseconds to complete. However, it is not a reliable mechanism for
     * canceling tests because a running test might block the application thread and
     * thus prevent the scheduled cancellation.
     * @since v18.0.0, v16.17.0
     * @param [name='The name'] The name of the test, which is displayed when reporting test results.
     * @param options Configuration options for the test. The following properties are supported:
     * @param [fn='A no-op function'] The function under test. The first argument to this function is a {@link TestContext} object. If the test uses callbacks, the callback function is passed as the
     * second argument.
     * @return Resolved with `undefined` once the test completes, or immediately if the test runs within {@link describe}.
     */
    function test(name?: string, fn?: TestFn): Promise<void>;
```